src/Lachain.Storage/Trie/NodeSerializer.cs

```diff
@@ -12,51 +12,101 @@ public static byte[] ToBytes(IHashTrieNode node)
+                            childId >>= 8;
+                            sz++;
+                        }
+                    }
+                    byte[] childByte = new byte[sz];
+                    uint idx = 0;
+                    for (int i = 0; i < _children.Length; i++)
+                    {
+                        ulong childId = _children[i];
+                        byte curSize = 0;
+                        while (childId > 0)
+                        {
+                            curSize++;
+                            childByte[idx + curSize] = (byte)(childId & 255);
+                            childId >>= 8;
+                        }
+                        childByte[idx] = curSize;
+                        idx += (uint)(curSize + 1);
+                    }
+                    return nodeByte.Concat(childByte).ToArray();
                 case LeafNode leafNode:
-                    return new byte[] {1}.Concat(leafNode.KeyHash).Concat(leafNode.Value).ToArray();
+                    return new byte[] { 1 }.Concat(leafNode.KeyHash).Concat(leafNode.Value).ToArray();
             }
 
             throw new InvalidOperationException($"Type {node.GetType()} is not supported");
         }
 
-        public static IHashTrieNode FromBytes(ReadOnlyMemory<byte> bytes)
+        public static IHashTrieNode FromBytes(byte[] bytes)
         {
             if (bytes.Length < 1)
             {
                 throw new ArgumentException("Empty bytes to deserialize");
             }
-            var type = bytes.Span[0];
+            var type = bytes[0];
             switch (type)
             {
                 case 0:
                     if (bytes.Length < 1 + 4)
                     {
                         throw new ArgumentException($"Not enough bytes to deserialize: {bytes.Length}");
                     }
-                    var mask = bytes.Slice(1, 4).Span.ToUInt32();
-                    var len = (int) BitsUtils.Popcount(mask);
-                    if (bytes.Length < 1 + 4 + 32 + len * 8)
+                    UInt32 mask = 0;
+                    UInt32 mul = 1;
+                    for (var i = 1; i <= 4; i++)
                     {
-                        throw new ArgumentException($"Not enough bytes to deserialize: {bytes.Length}");
+                        mask += mul * bytes[i];
+                        if (i < 4) mul = mul * 256;
                     }
-                    var hash = bytes.Slice(1 + 4, 32).ToArray();
-                    return new InternalNode(
-                        mask,
-                        Enumerable.Range(0, len)
-                            .Select(i => bytes.Slice(1 + 4 + 32 + i * 8, 8).Span.ToUInt64()),
-                        hash
-                    );
+                    var len = (int)BitsUtils.Popcount(mask);
+                    byte[] hash = new byte[32];
+                    for (var i = 0; i < 32; i++)
+                    {
+                        hash[i] = bytes[i + 5];
```

Comment:
It is better to check if this bytes are available (check the `bytes` length before the cycle)